src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1131,6 +1131,27 @@ ss::future<cluster::init_tm_tx_reply> tx_gateway_frontend::do_init_tm_tx(
     }
     auto term = term_opt.value();
 
+    auto cfg = _metadata_cache.local().get_topic_cfg(model::tx_manager_nt);
+    if (!cfg) {
+        co_return init_tm_tx_reply{tx_errc::partition_not_exists};
+    }
+    if (stm->hosted_transactions_inited()) {
+        if (!stm->check_tx_membership(tx_id)) {
+            co_return tx_errc::not_coordinator;
+        }
```

Comment:
With https://github.com/redpanda-data/redpanda/pull/10181/files#r1183242107 all public api (add_partition, add_offset, init_tx, end_txn) should expect hosted_transactions_inited and check if stm hosts given tx_id. Please think about fetch_tx. I can't decide on the spot.